### PR TITLE
MLOP-642 Document migration in Butterfree

### PR DESCRIPTION
**Pull request by roelschr**
## Why? :open_book:
After adding the new tool for schema migration, we must add some initial notes about it on our documentation.

## What? :wrench:
- Adding `cli.md` describing more about the new tool.
- Run update-docs created and updated docs that were outdated.

## Type of change
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Release

### CODE REVIEW

File: docs/source/cli.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Command-line Interface (CLI)
+
+Butterfree has now a command-line interface, introduced with the new automatic migration ability.
+
+As soon as you install butterfree, you can check what's available through butterfree's cli with:
+
+```shell
+$~ butterfree help
```
Comment (moromimay) [inline docs/source/cli.md:8]:
```suggestion
$~ butterfree --help
```

File: docs/source/cli.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+to come into your Feature Store. Normally, when creating a new feature set, you needed
+to manually create a new table in cassandra. Or, when creating a new feature in an existing
+feature set, you needed to create new column in cassandra too.
+
+Now, you can just use `butterfree migrate apply ...`, butterfree will scan your python
+files, looking for classes that inherit from `butterfree.pipelines.FeatureSetPipeline`,
+then compare its schema with the database schema where the feature set would be written.
+Then it will prepare migration queries and run against the databases.
+
+For more information, please, check `butterfree migrate apply help` :)
```
Comment (moromimay) [inline docs/source/cli.md:23]:
```suggestion
For more information, please, check `butterfree migrate apply --help` :)
```

**Review by moromimay — CHANGES_REQUESTED** [review]
Some questions and advice: :wink: 

- The `butterfree._cli` module, were you unable to map it? :thinking: 
- Requirements: we have the requirements in the path docs, we can create a make command to install them more easily, WDYT?
- The `cli.md` markdown file, I believe it needs to be referenced in `index.rst` or `home.md`, both to appear in the documentation and to make it easier for users to find the content.
- It would be interesting to put the supported databases in the AutomateMigration documentation, for example: cassandra.

**Comment by roelschr** [comment]
> Some questions and advice: 
> 
> * The `butterfree._cli` module, were you unable to map it? :thinking: 

I couldn't find a way to map it, but thinking better about it, does it make sense to show API documentations for a private interface that users shouldn't be using?

> * Requirements: we have the requirements in the path docs, we can create a make command to install them more easily, WDYT?

I guess that they should've been added into our requirements-dev file, that is installed locally for development. This PR does that, and now, it is easily installed by the `make requirements` command.

> * The `cli.md` markdown file, I believe it needs to be referenced in `index.rst` or `home.md`, both to appear in the documentation and to make it easier for users to find the content.

Done! Thanks :smile: 

> * It would be interesting to put the supported databases in the AutomateMigration documentation, for example: cassandra.

Done! Thanks :smile:

**Review by moromimay — APPROVED** [review]

**Pull request merged**